### PR TITLE
Declare missing dependency on python3-importlib-resources

### PR DESCRIPTION
**Pull request by cottsay**
Requires ros/rosdistro#28001.

Used here:
https://github.com/ros2/sros2/blob/74da58f7433833562226f1cf05113229cee11005/sros2/sros2/policy/__init__.py#L19-L25

The source code supports using the `importlib_resources` package, but on all of the platforms we're running regular builds for, we have ~~Python 3.7~~ Python 3.9, which provides the package as part of Python itself.

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
thanks for the fix.

This seems to introduce pip dependencies on some platforms listed in REP-2000 (Debian Buster).
Does this mean that sros2 will not be package-able on those platforms after this fix ?

**Comment by cottsay** [comment]
> Does this mean that sros2 will not be package-able on those platforms after this fix?

It appears that is the case. It actually won't matter, though, because `ament_package` itself requires `importlib_resources`, so nothing is going to work until the package is available.

I wasn't aware of this - I'll bring it up at the next ROS 2 meeting and we'll discuss the impact this will have on Galactic. You can hold this PR if you like, but there are a few other packages in `ros2.repos` that are already taking this dependency.

**Comment by mikaelarguedas** [comment]
> It appears that is the case. It actually won't matter, though, because ament_package itself requires importlib_resources, so nothing is going to work until the package is available.

Fair enough, :+1: for this then and we'll see if a follow-up is needed based on the result of the ROS 2 meeting discussion

File: sros2/package.xml [hunk 1]
```diff
@@ -16,11 +16,13 @@
   <exec_depend>ament_index_python</exec_depend>
   <exec_depend>python3-lxml</exec_depend>
   <exec_depend>python3-cryptography</exec_depend>
+  <exec_depend>python3-importlib-resources</exec_depend>
 
   <test_depend>ament_copyright</test_depend>
   <test_depend>ament_flake8</test_depend>
   <test_depend>ament_mypy</test_depend>
   <test_depend>ament_pep257</test_depend>
+  <test_depend>python3-importlib-resources</test_depend>
```
Comment (mikaelarguedas) [inline sros2/package.xml:25]:
```suggestion
```
From the [docs](http://wiki.ros.org/catkin/package.xml#Build.2C_Run.2C_and_Test_Dependencies) "Test Dependencies specify only additional dependencies for unit tests. They should never duplicate any dependencies already mentioned as build or run dependencies."

**Review by mikaelarguedas — CHANGES_REQUESTED** [review]

File: sros2/package.xml (hunk 1 above)
Comment (mikaelarguedas) [reply sros2/package.xml:25]: I went ahead and merged this suggestion. I'll let another @ros2/security-reviewers give the second approval

**Comment by cottsay** [comment]
> I'll bring it up at the next ROS 2 meeting and we'll discuss the impact this will have on Galactic

In the ROS 2 meeting we decided to target Bullseye for a variety of reasons, `importlib_resources` among them. It's also worth noting that we haven't built any binary packages for Debian in ROS 2 to-date.

**Review by mikaelarguedas — APPROVED** [review]

**Comment by mikaelarguedas** [comment]
> importlib_resources among them. It's also worth noting that we haven't built any binary packages for Debian in ROS 2 to-date.

:+1:, it's still good to know we're not standing in the way of people that may want to build debs for their own applications

**Comment by kyrofa** [comment]
> The source code supports using the importlib_resources package, but on all of the platforms we're running regular builds for, we have Python 3.7, which provides the package as part of Python itself.

Post-Foxy, which supported platforms use a Python prior to 3.7?

**Comment by cottsay** [comment]
My initial comment was incorrect - [it was introduced in Python 3.9](https://pypi.org/project/importlib-resources/), so anything earlier than that will need the package. Which is pretty much all supported platforms.

**Comment by kyrofa** [comment]
Ahhh, now I'm less confused :) .

**Review by kyrofa — APPROVED** [review]

**Pull request merged**